homeassistant/components/epic_games_store/calendar.py

```diff
@@ -0,0 +1,110 @@
+"""Calendar platform for a Epic Games Store."""
+
+from __future__ import annotations
+
+from collections import namedtuple
+from datetime import datetime
+from typing import Any
+
+from homeassistant.components.calendar import CalendarEntity, CalendarEvent
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import CalendarType, EGSCalendarUpdateCoordinator
+
+DateRange = namedtuple("DateRange", ["start", "end"])
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the local calendar platform."""
+    coordinator: EGSCalendarUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    entities = [
+        EGSCalendar(coordinator, entry.entry_id, CalendarType.FREE),
+        EGSCalendar(coordinator, entry.entry_id, CalendarType.DISCOUNT),
+    ]
+    async_add_entities(entities, True)
```

Comment:
```suggestion
    async_add_entities(entities)
```
You're using a coordinator so data is already present when you add entities, no need to do another update